### ISSUE TITLE
Keyboard in going over the page view - on android

### ISSUE THREAD

**Issue by shabib3** [NativeScript/NativeScript#5088]
### Did you verify this is a real problem by searching [Stack Overflow]
There is some opened issues about it but no one of them give me a full solution...
The issues that I found:
* [Keyboard overlapping textview - discourse nativescript](https://discourse.nativescript.org/t/keyboard-overlapping-textview/3213)
* [Keyboard overlapping over textfield - Telerik forum](https://www.telerik.com/forums/keyboard-overlapping-over-textfield)

### Tell us about the problem
When I am in a page (with ScrollView), and I click any TextField I would expect the keyboard to show up and the page to shrink, and still gets the scrolling functionalities all over the page...But the thing that actually happens is that the keyboard is overlaps the view and won't let me scroll down...The weird thing here is when I put the app on background and then come back to it is works just as it should, the page is scrolling as expect.

Here is my video example for the issue -> [Nativescript keyboard overlap screen bug](https://www.youtube.com/watch?v=MbmAn7J0jV0)
In the video you can see that I go the "Create Account" page and start to fill up the fields and when I want to go to the "Password" field I cannot scroll down unless I am closing the keyboard and then pressing the "Password" field again...But when I putting the app on background and then return back in, the scrolling works just fine!

Something that may help - You can notice in the video that in first time that I run the app before I put in in background when the issue is happening, when I close the keyboard and pressing the "Password" TextField the all page in scrolling down - **including the ActionBar**. But when I put the app on background and come back to it, the scrolling is works just fine only on the ScrollView and the ActionBar in stand steady.

### Which platform(s) does your issue occur on?
Android

### Please provide the following version numbers that your issue occurs with:
- CLI: 3.3.0
- Cross-platform modules: 3.3.0
- Runtime(s): "tns-android": {
      "version": "3.4.0-2017-10-31-1"
    }
***(tns platform add android@next)

Thank you very much! :)

**Comment by NickIliev** [comment]
@shabib3 thank you for logging this issue. I have logged this one as a bug [here](https://github.com/NativeScript/NativeScript/issues/5086) so I am closing this one in favor of the linked issue.

**Comment by shabib3** [comment]
So after I saw the open bug that @NickIliev opened, I not sure that there is the same situation...Maybe the bug fix will fix both, but just so you know there is a chance that this is not the same issue...

When I first opened this issue I didn't have a "returnKeyType='next'" on the TextFields and after I added "returnKeyType='next'" it still not working as it should work, but now I can get to the "Password" TextField (the last one) straight from the "Email" TextField (second to last) by pressing enter...But it is still not ok because it is scrolling the Page down instead of the ScrollView...And still if I put the app on background and then return back to it, it looks like working as it should.

I am adding a new example video -> [Nativescript keyboard overlap screen bug 2](https://www.youtube.com/watch?v=2gTUSQ8yyDk)

And my html code:
```
<ActionBar title="Sign Up"></ActionBar>
<AbsoluteLayout>
    <ScrollView class="dark-background full-stretch">
        <StackLayout>
            <StackLayout class="field-parent">
                <StackLayout>
                    <Label text="Name" class="field-label"></Label>
                    <GridLayout columns="*, *">
                        <TextField returnKeyType="next" class="field-text-area" col="0" [hint]="strings.get('firstNameSignUp')" [(ngModel)]="firstName"></TextField>
                        <TextField returnKeyType="next" class="field-text-area" col="1" [hint]="strings.get('lastNameSignUp')" [(ngModel)]="lastName"></TextField>
                    </GridLayout>
                </StackLayout>
                <StackLayout>
                    <Label text="Email Address" class="field-label"></Label>
                    <GridLayout columns="*, auto">
                            <StackLayout col="0">
                                <TextField returnKeyType="next" class="field-text-area" [hint]="strings.get('emailSignUp')" keyboardType="email" [(ngModel)]="email"></TextField>
                            </StackLayout>
                            <StackLayout col="1" class="username-valid-state">
                                <Label *ngIf="email && !_inputValidatorService.validateEmail(email)" class="icon bad-icon" text="&#xf057;"></Label>
                                <Label *ngIf="email && _inputValidatorService.validateEmail(email)" class="icon check-icon" text="&#xf058;"></Label>
                            </StackLayout>
                        </GridLayout>
                </StackLayout>
                <StackLayout>
                    <StackLayout orientation="horizontal">
                        <Label text="Password" class="field-label"></Label>
                        <Label [text]="passwordSecureMode ? 'Show' : 'Hide'" class="show-hide-text" (tap)="secureModeChange()"></Label>
                    </StackLayout>
                    <GridLayout columns="*, auto">
                        <StackLayout col="0">
                            <TextField class="field-text-area" [hint]="strings.get('passwordSignUp')" [secure]="passwordSecureMode" [(ngModel)]="password"></TextField>
                        </StackLayout>
                        <StackLayout col="1" class="username-valid-state">
                            <Label *ngIf="password && !_inputValidatorService.validatePassword(password)" class="icon bad-icon" text="&#xf057;"></Label>
                            <Label *ngIf="password && _inputValidatorService.validatePassword(password)" class="icon check-icon" text="&#xf058;"></Label>
                        </StackLayout>
                    </GridLayout>
                    <Label class="password-rules" text="*** Password must have at least 8 characters, one capital letter, one lowercase letter and one number" textWrap="true"></Label>
                </StackLayout>
            </StackLayout>
        </StackLayout>
    </ScrollView>

    <StackLayout class="float-button-mainly-container">
        <floating-action-button (tap)="nextToCreateProfile()" text="&#xf061;" buttonBackgroundColor="#f2f2f2" textColor="#333333" buttonTapColor="#cccccc"
            [inValidCallback]="isValidForm"></floating-action-button>
    </StackLayout>
</AbsoluteLayout>
```

Hope that will help!
Here for and question -> user name at slack is "shabib"

**Comment by erkanarslan** [comment]
I also have the same problem. When I put app in background and bring it back it works correctly. But also it works correctly when I open and close side drawer. The problem is that when keyboard opened, page content is scrolled up until text field is visible. But when app comes back from background, or side drawer  is opened and closed page is resized to fit all content.

I am not an Android developer but I think it is related to `android:windowSoftInputMode`. Look at the last two items in the table in [this link](https://developer.android.com/guide/topics/manifest/activity-element.html#wsoft).

Here what it says for `adjustPan`:

> The activity's main window is not resized to make room for the soft keyboard. Rather, the contents of the window are automatically panned so that the current focus is never obscured by the keyboard and users can always see what they are typing. **This is generally less desirable than resizing, because the user may need to close the soft keyboard to get at and interact with obscured parts of the window.**

This explains the problem @shabib3 is facing.

I believe that if we set this setting to `adjustResize` this problem might be fixed.

**Comment by shabib3** [comment]
@erkanarslan 
It looks interesting and we should try it...did you try it? do you know how can we change all this parameters you mentioned?

Thanks!

**Comment by erkanarslan** [comment]
I will try it this evening and share the results.

On Dec 21, 2017 19:15, "shabib3" <notifications@github.com> wrote:

@erkanarslan <https://github.com/erkanarslan>
It looks interesting and we should try it...did you try it? do you know how
can we change all this parameters you mentioned?

Thanks!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/NativeScript/NativeScript/issues/5088#issuecomment-353391228>,
or mute the thread

**Comment by shabib3** [comment]
@erkanarslan 
I am happy to tell you that you are the KING of the WORLD!!!
WORKS LIKE HEAVEN!!!

I am really thankful!!!

**Comment by shabib3** [comment]
In the `AndroidManifest.xml` file just add right after the `android:theme` value the following value ->
`android:windowSoftInputMode="adjustResize"`

**Comment by erkanarslan** [comment]
@shabib3 You are welcome. I am glad that this problem is solved.

@NickIliev It seems that this is not a problem that needs to be solved in Nativescript but it is just a configuration that we need to add it to Android manifest file when we need it. We can close this issue.

**Comment by weijyewang** [comment]
The solutions did move the input fields but the button beneath the form fields will still be blocked by the keyboard. Did any of you managed to adjust the button position as well when the keyboard came up?

**Comment by bhavincb** [comment]
hii @NickIliev ,
i am also having the same issue. and tried to modify AndroidManifest.xml file and added `android:windowSoftInputMode="adjustResize"` but running `tns debug android --clean` override the file so changes doesn't apply.
so what is the correct approach?
 also note that i'm also having issue when i'm using telerik SideDrawer as discussed at [https://discourse.nativescript.org/t/keyboard-overlapping-textview/3213/11](url)

**Comment by waszqba** [comment]
@bhavincb If something's not willing to apply, I get the best success rate with `tns platform remove <ios|android>` && `tns platform add <ios|android>`

**Comment by bhavincb** [comment]
hii @waszqba , thankx for suggestion. but i had already done that multiple times. it is still not working. keyboard overlap the view

**Comment by morzyns** [comment]
Same issue here with `android:windowSoftInputMode="adjustResize"` but running `tns debug android --clean`
with  tns-android: 3.4.1

**Comment by skibrit** [comment]
same issue here. android:windowSoftInputMode="adjustResize" didn't work for me. android:windowSoftInputMode="adjustPan" worked but it created a new problem. Now keyboard pushes some page content outside the screen including the action bar.

**Comment by racknoris** [comment]
Is it possible that with:
`nativescript 4.0.0`, `tns-core-module 4.0.0`, `tns-android 4.0.1`, the behaviour of the keyboard works as it should on default?

**Comment by racknoris** [comment]
@Guatom Same here, my settings were messed up, so later on i saw that i must use `adjustResize` or `adjustPan`

**Comment by Guatom** [comment]
@racknoris and it is important that the value must be at `activity` level and not at `Application` level. My first try went wrong because somewhere it says that the value must be after `android:theme`, and that appears twice in my `AndroidManifest.xml`.

**Comment by racknoris** [comment]
Same thing happened to me, 
In my case, i have multiple `activity` tags, so i placed those values inside all of them

**Comment by kono-paku** [comment]
I think iOS have the same issue. Any idea to solve?

**Comment by racknoris** [comment]
@kono-paku Yes- use https://www.npmjs.com/package/nativescript-iqkeyboardmanager, it works like magic

**Comment by xeroxstar** [comment]
I have the same problem when TextField is focused, the keyboard overlapping the text. I tried everything, i added a android:windowSoftInputMode="adjustResize" and android:windowSoftInputMode="adjustPan" and it's not working.

**Comment by webleaf** [comment]
Don't work for me. Android. NS4.
It work for textfields in `RadDataForm` even without `android:windowSoftInputMode="adjustResize"`
But don't work for me just for `TextView` with an without `adjustResize`
How does it implemented for `RadDataForm`?

**Comment by webleaf** [comment]
Looks like it don't work without `ScrollView`. With `ScrollView` it works even without `adjustResize`.

**Comment by juanreynolds** [comment]
Same here. With non-angular template it works, with angular template it does not in portrait orientation.

**Comment by bearoutthere** [comment]
My layout is using tabs.  Putting scrollviews inside tabs did not work with adjustResize but did work with adjustPan.  adjustPan does push the actionbar off the screen but i think it's a feature not a bug, since it clearly shows the user that they have to finish entering data before doing anything else.  The mention of these settings should be added to Nativescript softkeyboard documentation.

**Comment by BMwanza** [comment]
@bearoutthere Are is using a tab view the issue then? This is a really annoying issue. I have tried all the above solutions but none work for me

**Comment by phenric** [comment]
To whom with the same issue, I suggest you to check this post [https://github.com/NativeScript/nativescript-angular/issues/1277#issuecomment-381576104](https://github.com/NativeScript/nativescript-angular/issues/1277#issuecomment-381576104).

It's related to the solution @shabib3 gave,

> In the `AndroidManifest.xml` file just add right after the `android:theme` value the following value ->
> `android:windowSoftInputMode="adjustResize"`

with an auto-resizing subtlety when you hit the "next" button,
`android:windowSoftInputMode="stateHidden|adjustResize"`

As mentioned several times, you have to put this line within the _activity_ just after the _android:theme_ into _AndroidManifest.xml_

It worked for me

**Comment by csimpi** [comment]
Doesn't work for me :(

**Comment by mariusfilo** [comment]
https://github.com/NativeScript/nativescript-angular/issues/1277

**Comment by Alexander-Bliznyuk** [comment]
By default soft input mode is unspecified. 
According to [Android API reference](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#SOFT_INPUT_ADJUST_UNSPECIFIED)
> the system will try to pick one or the other depending on the contents of the window

Also, `adjustResize` is now [deprecated](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#SOFT_INPUT_ADJUST_RESIZE).
.

**Comment by Peliukas** [comment]
In my case `android:windowSoftInputMode="adjustPan"` was the solution

**Comment by naseerhussain** [comment]
None of the above solutions worked for me

**Comment by EvidenceEkanem** [comment]
None of the above solutions worked for me, please I need help for Nativescript 8.9.3
